### PR TITLE
Shrinkwrap auditing DESKTOP-1771

### PR DESCRIPTION
**Pull request by awendland**
Checks that all dependencies in `package.json` also exist in `npm-shrinkwrap.json`, and that their versions are the same (this should detect and prevent the RN 0.31 update issue).

### CODE REVIEW

**Comment by awendland** [comment]
👀 @keybase/react-hackers 

You can see some example output of this script breaking CI here https://ci.keybase.io/job/client/job/PR-4055/2/execution/node/204/log/

**Comment by cjb** [comment]
r? @jzila too

:thumbsup: 

Any idea how long this takes?  If it takes a few minutes, does it make the total execution time for a run that much longer?  If it did, we'd perhaps make it happen in parallel to test_linux.

**Comment by awendland** [comment]
@cjb On my local machine it takes ~3-5 seconds. I'm not sure how to profile it on Jenkins.

**Comment by cjb** [comment]
@awendland That sounds okay to me.  It probably doesn't take longer on Jenkins.

File: Jenkinsfile [hunk 1]
```diff
@@ -133,6 +133,8 @@ if (env.CHANGE_TITLE && env.CHANGE_TITLE.contains('[ci-skip]')) {
                                             sh "npm run vendor-install"
                                             sh "unzip ${env.KEYBASE_JS_VENDOR_DIR}/flow/flow-linux64*.zip -d ${env.BASEDIR}"
                                             sh "${env.BASEDIR}/flow/flow status shared"
+                                            sh "npm run shrinkwrap-audit"
```
Comment (jzila) [inline Jenkinsfile]: can this happen in parallel with flow? Is it dependent on vendor-install? If it isn't dependent, can do:

```groovy
parallel (
    test_linux_js_flow: {
        dir("desktop") {
            <flow stuff here...>
        }
    },
    test_linux_js_shrinkwrap: {
        sh "npm run shrinkwrap-audit -- ./desktop"
        sh "npm run shrinkwrap-audit -- ./react-native"
    },
)
```

Comment (jzila) [inline Jenkinsfile]: Obviously if this is the case, `shrinkwrap-audit.js` will have to be in the root directory instead of desktop.

Comment (awendland) [inline Jenkinsfile]: The script is dependent on the `babel-node` package (and related helpers) that are installed in `desktop/` by `npm run vendor-install`, I believe.

Comment (jzila) [inline Jenkinsfile]: OK. Can still run it in parallel once `vendor-install` happens though.

**Comment by jzila** [comment]
Actually, it doesn't seem to be slowing down builds. Go ahead and ship.

**Pull request merged**